opnode/node/node.go

```diff
@@ -16,139 +13,126 @@ import (
-		// TODO: if we start from a squashed snapshot we might have a non-zero L2 genesis number
-		L2: eth.BlockID{Hash: conf.L2Hash, Number: 0},
-	}
-}
-
-type OpNodeCmd struct {
-	L1NodeAddrs   []string `ask:"--l1" help:"Addresses of L1 User JSON-RPC endpoints to use (eth namespace required)"`
-	L2EngineAddrs []string `ask:"--l2" help:"Addresses of L2 Engine JSON-RPC endpoints to use (engine and eth namespace required)"`
-
-	LogCmd `ask:".log" help:"Log configuration"`
-
-	Genesis GenesisConf `ask:".genesis" help:"Genesis anchor point"`
-
-	// during later sequencer rollup implementation:
-	// TODO: multi-addrs option (static peers)
-	// TODO: bootnodes option (bootstrap discovery of more peers)
-
-	log log.Logger
-
-	// (combined) source to fetch data from
-	l1Source eth.L1Source
+type Config struct {
+	// L1 and L2 nodes
+	L1NodeAddrs   []string // Addresses of L1 User JSON-RPC endpoints to use (eth namespace required)
+	L2EngineAddrs []string // Addresses of L2 Engine JSON-RPC endpoints to use (engine and eth namespace required)
 
-	// engines to keep synced
-	l2Engines []*driver.EngineDriver
+	// Genesis Information
+	L2Hash common.Hash // Genesis block hash of L2
+	L1Hash common.Hash // Block hash of L1 after (not incl.) which L1 starts deriving blocks
+	L1Num  uint64      // Block number of L1 matching the l1-hash
 
-	l1Downloader l1.Downloader
-
-	ctx   context.Context
-	close chan chan error
+	LogCfg LogConfig
 }
 
-func (c *OpNodeCmd) Default() {
-	c.L1NodeAddrs = []string{"http://127.0.0.1:8545"}
-	c.L2EngineAddrs = []string{"http://127.0.0.1:8551"}
+// Check verifies that the given configuration makes sense
+func (cfg *Config) Check() error {
+	err := cfg.LogCfg.Check()
+	if err != nil {
+		return fmt.Errorf("Error checking log sub-config: %w", err)
+	}
+	return nil
 }
 
-func (c *OpNodeCmd) Help() string {
-	return "Run optimism node"
+type OpNode struct {
+	log          log.Logger
+	l1Source     eth.L1Source           // (combined) source to fetch data from
+	l1Downloader l1.Downloader          // actual downloader
+	l2Engines    []*driver.EngineDriver // engines to keep synced
+	ctx          context.Context        // Embeded CTX to be removed
+	done         chan struct{}
 }
 
-func (c *OpNodeCmd) Run(ctx context.Context, args ...string) error {
-	logger := c.LogCmd.Create()
-	c.log = logger
-	c.ctx = ctx
-
-	if c.Genesis == (GenesisConf{}) {
-		return errors.New("genesis configuration required")
+func (conf *Config) GetGenesis() rollup.Genesis {
+	return rollup.Genesis{
+		L1: eth.BlockID{Hash: conf.L1Hash, Number: conf.L1Num},
+		// TODO: if we start from a squashed snapshot we might have a non-zero L2 genesis number
+		L2: eth.BlockID{Hash: conf.L2Hash, Number: 0},
 	}
+}
 
-	l1Sources := make([]eth.L1Source, 0, len(c.L1NodeAddrs))
-	for i, addr := range c.L1NodeAddrs {
+func New(ctx context.Context, cfg *Config) (*OpNode, error) {
+	if err := cfg.Check(); err != nil {
+		return nil, err
+	}
+	log := cfg.LogCfg.NewLogger()
```

Comment:
It would be neat if we can attach a test-logger instance to the node during testing, instead of creating a logger from the log-config.